cluster_agent/jobbergate/api.py

```diff
@@ -65,19 +66,31 @@ async def mark_as_submitted(job_submission_id: int, slurm_job_id: int):
         response.raise_for_status()
 
 
-async def notify_submission_rejected(
-    params: DoExceptParams, job_submission_id: int
-) -> None:
+@dataclass
+class NotifySubmission:
```

Comment:
I'm not sure that I'm seeing any great advantage for using a class for this instead of just defining a function to do it. The biggest reason I'm not sure about this is that we are naming a class with a verb here and typically, classes should be named as objects. What do you think?

Reply:
The problem here is that we need to somehow store `job_submission_id` and keep the function's signature as expected by `handle_errors_async`. The previous approach with functions and `functools.partial` didn't work asynchronously, so I changed to this class. I'm ok with renaming it, what do you think of `SubmissionNotifier`?

Reply:
Solved in [8e5e7d5](https://github.com/omnivector-solutions/cluster-agent/pull/48/commits/8e5e7d52534d8e5d85a688ae37e0eb7388b95da7)